### ISSUE TITLE
Allow ES2015 Module Syntax

### ISSUE THREAD

**Issue by reubenlillie** [11ty/eleventy#560]
Is there a way to use the native ES2015 module syntax with `import * from`, `export`, `export default`, and such rather than the CommonJS `require()` and `module.exports`?

If not, can we make a way?

**Comment by kleinfreund** [comment]
Node does not have stable support for ES modules. I think this has to wait until that’s out of the oven.

**Comment by zachleat** [comment]
It’s a fair question! I’m reluctant to add something like Babel for this. I wonder if it works at all with node’s `--experimental-modules` flag? https://nodejs.org/api/esm.html

Would this mean something like `.11tydata.mjs` and `.11ty.mjs` extensions?

**Comment by jevets** [comment]
I'd also welcome the change.

I've used `--experimental-modules` for Eleventy projects and it does work fine w/o having to use `.mjs` extensions, but this could be dependent on the node version. (I usually stick w/ `require()` to keep things simpler across environments.)

**Comment by hcavalieri** [comment]
@jevets how do you use `--experimental-modules` with Eleventy? I can't seem to find the right way to add it to my scripts :(

**Comment by jevets** [comment]
I've used it in npm package.json scripts, something like this (from [my human] memory). 

```json
// package.json
{
  "scripts": {
    "dev": "npm run eleventy --serve --experimental-modules"
  }
}
```

I think I tried a slew of combinations until it worked, can try to dig up a project if you can't get it working

**Comment by reubenlillie** [comment]
@jevets, I want to believe your solution is possible, but I haven't quite figured out how to make it work myself. Do you have a project repo you can share with a script that does not require `(.11ty).mjs` files?

**Comment by valtlai** [comment]
Node 13.2.0 supporting ES modules by default (i.e. without the flag) was just [released](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V13.md#13.2.0). Eleventy doesn’t work with `"type": "module"` on `package.json`.

**Comment by zachleat** [comment]
Hey y’all, moving this discussion to #836

Related work for `.cjs` in Eleventy is available in #826 and #837

Please follow along at those individual issues—thanks!
